include/term-slug.php

```diff
@@ -0,0 +1,179 @@
+		 *
+		 * @since 3.3
+		 *
+		 * @param PLL_Language|null $lang     Found language object, null otherwise.
+		 * @param string            $taxonomy Term taxonomy.
+		 * @param string            $slug     Term slug
+		 */
+		$lang = apply_filters( 'pll_inserted_term_language', null, $this->taxonomy, $this->slug );
+		if ( ! $lang instanceof PLL_Language ) {
+			return false;
+		}
+		$this->lang = $lang;
+
+		$this->parent = 0;
+		if ( is_taxonomy_hierarchical( $this->taxonomy ) ) {
+			/**
+			 * Filters the subsequently inserted term parent.
+			 *
+			 * @since 3.3
+			 *
+			 * @param int          $parent   Parent term ID, 0 if none.
+			 * @param string       $taxonomy Term taxonomy.
+			 * @param string       $slug     Term slug
+			 */
+			$this->parent = apply_filters( 'pll_inserted_term_parent', 0, $this->taxonomy, $this->slug );
+
+			$this->slug .= $this->maybe_get_parent_suffix();
+		}
+
+		if ( ! $this->slug ) {
+			if ( $this->model->term_exists( $this->name, $this->taxonomy, $this->parent, $this->lang ) ) {
+				// Returns the current empty slug if the term exists with the same name and an empty slug.
+				// Same as WP does when providing a term with a name that already exists and no slug.
+				return false;
+			} else {
+				$this->slug = sanitize_title( $this->name );
+			}
+		}
+
+		if ( ! term_exists( $this->slug, $this->taxonomy ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns the parent suffix for the slug only if parent slug is the same as the given one.
+	 * Recursively appends the parents slugs like WordPress does.
+	 *
+	 * @since 3.3
+	 * @since 3.7 Moved from `PLL_Share_Term_Slug`to `PLL_Term_Slug`.
+	 *
+	 * @return string Parents slugs if they are the same as the child slug, empty string otherwise.
+	 */
+	private function maybe_get_parent_suffix() {
+		$parent_suffix = '';
+		$the_parent    = get_term( $this->parent, $this->taxonomy );
+
+		if ( ! $the_parent instanceof WP_Term || $the_parent->slug !== $this->slug ) {
+			return $parent_suffix;
+		}
+
+		/**
+		 * Mostly copied from {@see wp_unique_term_slug()}.
+		 */
+		while ( ! empty( $the_parent ) ) {
+			$parent_term = get_term( $the_parent, $this->taxonomy );
+			if ( ! $parent_term instanceof WP_Term ) {
+				break;
+			}
+			$parent_suffix .= '-' . $parent_term->slug;
+			if ( ! term_exists( $this->slug . $parent_suffix ) ) {
+				break;
+			}
+			$the_parent = $parent_term->parent;
+		}
+
+		return $parent_suffix;
+	}
+
+	/**
+	 * Returns the term slug, suffixed or not.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string $separator The separator for the slug suffix, or empty.
+	 * @return string
+	 */
+	public function get( string $separator = '' ): string {
+		if ( empty( $separator ) ) {
+			return $this->slug;
+		}
+
+		return $this->slug . $separator . $this->lang->slug;
+	}
+
+	/**
+	 * Returns the existing term ID.
+	 *
+	 * @since 3.7
+	 *
+	 * @return int
+	 */
+	public function get_term_id(): int {
```

Comment:
I think it's better to keep it to get the existing term by slug since we need the parent, see [here](https://github.com/polylang/polylang/pull/1490#discussion_r1647713347).